src/main/java/net/sf/jabref/exporter/BibDatabaseWriter.java

```diff
@@ -354,6 +356,27 @@ private void writeString(Writer fw, BibtexString bs, Map<String, BibtexString> r
         fw.write("}" + Globals.NEWLINE);
     }
 
+    private void writeUserCommentsForString(BibtexString string, Writer out) throws IOException {
+        String parsedSerialization = string.getParsedSerialization();
+
+        if(parsedSerialization != null) {
+
+            try {
+                // get the text before the string
+                String prolog = string.getParsedSerialization().substring(0, string.getParsedSerialization().indexOf('@'));
```

Comment:
Good idea, will do.

However, I am a little bit opposed to simplifying the roundtrip test. This is the discussion on whether we prefer unit over integration tests. We have been there before :)